src/frontend/js/components/RootSearchSuggestField/index.tsx

```diff
@@ -132,7 +132,7 @@ export const RootSearchSuggestField = ({
       onSuggestionsFetchRequested={async ({ value: incomingValue }) =>
         onSuggestionsFetchRequested(await getFilters(), setSuggestions, incomingValue)
       }
-      onSuggestionHighlighted={({ suggestion }) => setHasHighlightedSuggestion(!suggestion)}
+      onSuggestionHighlighted={({ suggestion }) => setHasHighlightedSuggestion(!!suggestion)}
```

Comment:
Is there a test that we can add that should have caught it?

Reply:
That's a point to investigate, from my point of view, the following test should already warn us about this mistake.
https://github.com/openfun/richie/blob/master/src/richie/apps/search/indexers/courses.py#L89

Reply:
Ok understood, the test passed because in fact with the bug, location.assign was called two times for this use case.
First time, by the `onSuggestionSelected` callback then by `onKeyDown` input trigger.

So, the first redirect (`en/courses/42`) exists but in reality there was a second redirect at the same time which redirected on `/en/courses/?limit=13&offset=0&query=Course%20%2342`.

Reply:
:muscle: